.travis.yml

```diff
@@ -10,4 +10,7 @@ env:
     - TOX_VERSION="tox-3.x"
 install:
     - pip install --constraint "$TOX_VERSION" .
-script: "tox"
+script:
+    - tox
+    # assure that we fail a build if it contains leftovers
+    - git diff --exit-code
```

Comment:
When could this happen in a Travis build?

Reply:
There are a zillion cases where this can happen: someone toucing the .gitignore file, someone adding a new command that leaves new leftovers after running it.

Here is one example: make a PR that removes the .tox from .gitignore and see what happes on Travis, it will pass. Mainly this like is regression protection.

Reply:
But un-tracked files won't cause an issue any way:

```
dcrosta@host:~ $ git init x
Initialized empty Git repository in /Users/dcrosta/x/.git/
dcrosta@host:~ $ cd x
(master) dcrosta@host:~/x ± touch foo
(master) dcrosta@host:~/x ± git diff --exit-code
(master) dcrosta@host:~/x ± echo $?
0
(master) dcrosta@host:~/x ±
```

I can't imagine a case where a test changes a tracked file. I would certainly consider that an issue! But I think it's unlikely enough that we can probably skip this.

Reply:
@dcrosta sorry for the delay on fixing it. I think that current version works correctly now.